### ISSUE TITLE
feature request: <map to supported integers>

### ISSUE THREAD

**Issue by claudiofsr** [jmcnamara/polars_excel_writer#4]
### Feature Request

Good morning!!!

I Try to use
```
fn write_xlsx(df: &DataFrame) -> PolarsResult<()> {
    let mut xlsx_writer = PolarsXlsxWriter::new();

    xlsx_writer.write_dataframe(df)?;
    xlsx_writer.save("df_result.xlsx")?;

    Ok(())
}
```
And I get this Error:
```
Error: ComputeError(ErrString("Polars AnyValue data type 'i64' is not supported by Excel"))
```

Is it possible to map 'i64' and all unsupported integers to 'u32' supported by Excel?

**Comment by jmcnamara** [comment]
I can probably add a `i32::try_from()` and then  only panic/warn if the conversation fails. I’ll look into it.

**Comment by claudiofsr** [comment]
I try this:
```
// https://crates.io/crates/polars_excel_writer
fn write_xlsx(df: &DataFrame) -> Result<(), Box<dyn Error>> {
    let mut xlsx_writer = PolarsXlsxWriter::new();

    let df_formatted: DataFrame = to_supported_integers(df)?;

    xlsx_writer.write_dataframe(&df_formatted)?;
    xlsx_writer.save("df_consolidacao_natureza_da_bcalc_result.xlsx")?;

    Ok(())
}

/// Format to 'i64' and all unsupported integers to 'u32' supported by Excel
fn to_supported_integers(dataframe: &DataFrame) -> Result<DataFrame, Box<dyn Error>> {
    Ok(
        dataframe
            .clone()
            .lazy()
            .with_columns([
                all()
                .apply(format_to_u32, GetOutput::same_type())
            ])
            .collect()?
    )
}

pub fn format_to_u32(series: Series) -> Result<Option<Series>, PolarsError> {
    if filter_integer(series.dtype()) {
        Ok(Some(series.cast(&DataType::UInt32)?))
    } else {
        Ok(Some(series))
    }
}

fn filter_integer(t: &DataType) -> bool {
    use DataType::*;
    matches!(t, UInt64 | Int64)
}
```

**Comment by jmcnamara** [comment]
> I try this:

That looks like a good workaround. For my information how did you end up with i/u64 in your dataframe. Do you create it it from CSV or similar. I've seen some Polars integer interpretations to i/u64 by default.

**Comment by claudiofsr** [comment]
I open two csv files to cross-reference data. [join_with_assignments](https://github.com/claudiofsr/join_with_assignments/blob/master/src/lib.rs#L293)

I believe that by default Polars reads integers like I64.
I could have initially changed I64 to U32!

**Comment by claudiofsr** [comment]
So far I have added the truncate_series function.

```
/// Write Dataframe to xlsx Excel file
pub fn write_xlsx(df: &DataFrame, filename: &str) -> Result<(), Box<dyn Error>> {
    let filepath = &format!("{filename}.xlsx");
    println!("Write DataFrame to {filepath}\n");

    let mut xlsx_writer = PolarsXlsxWriter::new();

    let data_frame: DataFrame = format_to_excel(df)?;

    xlsx_writer.write_dataframe(&data_frame)?;
    xlsx_writer.save(filepath)?;

    Ok(())
}

/// Format data supported by Excel
fn format_to_excel(data_frame: &DataFrame) -> Result<DataFrame, Box<dyn Error>> {
    let df_formated: DataFrame = data_frame
        .clone()
        .lazy()
        .with_columns([
            all()
            .apply(format_data, GetOutput::same_type())
        ])
        .collect()?;
    
    Ok(df_formated)
}

/// Format DataType
fn format_data(series: Series) -> Result<Option<Series>, PolarsError> {
    match series.dtype() {
        DataType::UInt64 | DataType::Int64 => format_to_u32(series),
        DataType::Utf8 => truncate_series(series), // to_n_chars(series)
        _ => Ok(Some(series)),
    }
}

/// Try to format 'u64' or 'i64' integers to 'u32' supported by Excel.
fn format_to_u32(series: Series) -> Result<Option<Series>, PolarsError> {
    Ok(Some(series.cast(&DataType::UInt32)?))
}

fn truncate_series(series: Series) -> Result<Option<Series>, PolarsError> {
    let new_series: Series = series
        .utf8()?
        .par_iter() // rayon: parallel iterator
        .map(|option_str: Option<&str>|
            option_str.map(|s| truncate_string(s, 32767)) // 2 ^ 15
        )
        .collect::<Utf8Chunked>()
        .into_series();

    Ok(Some(new_series))
}

// https://stackoverflow.com/questions/38461429/how-can-i-truncate-a-string-to-have-at-most-n-characters
fn truncate_string(s: &str, max_chars: usize) -> &str {
    match s.char_indices().nth(max_chars) {
        None => s,
        Some((idx, _)) => &s[..idx],
    }
}
```

**Comment by jmcnamara** [comment]
I've dded support for writing `u64` and `i64` number within Excel's limitations. This implies a loss of precision outside Excel's integer range of +/-  999,999,999,999,999 (15 digits).

It is available in v0.5.0.
